### PR TITLE
Network: strict OpenAPI interface for API

### PR DESCRIPTION
**Pull request by reinkrul**
PoC of https://github.com/nuts-foundation/nuts-node/issues/1607, so we can decide whether we think this is better.

I personally like the RPC-style API functions more (because they are safer and more explicit), but the naming is bad (long request/response object names) leading to code that doesn't read very nice.

But it still prevents bugs where the API implementation would handle parameters/response codes/types differently than spec'd.

Test code is more or less the same, some ugly casting required but no more weird HTTP request/response recorders.

I'd 4-8 hours to rewrite all APIs and tests. When subtracting generated code from added LoCs, the LoCs with and without strict server interfaces (including tests) are more or less equal.

Toughts?

### CODE REVIEW

**Comment by gerardsn** [comment]
I like the strict definitions and structure this creates for a API call.

Some extra consideration related to this change
- CLI commands need to be updated accordingly
- New VC / DID are currently specced to use application/vc+json / application/did+json as content type in transactions, in oapi we defined something else, and implementation returns them as application/json on creation. (There seems to be no consensus on the mediatypes yet.) Using strict interfaces probably requires fixing this issue first.
- `/api/*` currently exists purely for comparing oapi generated structs with our internal struct. Would we be able to drop these tests when adopting the strict interfaces?

**Comment by woutslakhorst** [comment]
It seems that our code looks cleaner (at least less) plus it adds testability for the actual content-type. The echo.Context is gone, also a plus.

What is the default behaviour when param/body parsing results in an error? Is our middleware still returning a nice RFC style Problem?

File: network/api/v1/api_test.go [hunk 1]
```diff
@@ -328,52 +239,40 @@ func TestApiWrapper_Reprocess(t *testing.T) {
...
 func TestWrapper_Preprocess(t *testing.T) {
-	ctrl := gomock.NewController(t)
-	w := &Wrapper{}
-	ctx := mock.NewMockContext(ctrl)
-	ctx.EXPECT().Set(core.OperationIDContextKey, "foo")
-	ctx.EXPECT().Set(core.ModuleNameContextKey, "Network")
-
-	w.Preprocess("foo", ctx)
+	t.SkipNow()
+	// TODO: Test strict middleware
```
Comment (gerardsn) [inline network/api/v1/api_test.go:268]: is still needed now the Preprocessed is no longer generated?

File: test/http/api.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package http
```
Comment (gerardsn) [inline test/http/api.go:1]: license + some comments on the methods

**Review by gerardsn — CHANGES_REQUESTED** [review]

File: test/http/api.go (hunk 2 above)
Comment (reinkrul) [reply test/http/api.go:1]: I don't really care about comments on test funcs?

Comment (gerardsn) [reply test/http/api.go:19]: I'm still slow at reading code / processing new concepts. Oneliners telling me what to expect or when to use it help a lot. Having to go to a test function to figure out what it does requires moving through code and distracts from what you are trying to test.

Although these functions are probably simple enough to understand

**Review by gerardsn — APPROVED** [review]

**Pull request merged**